fuel/transformers/image.py

```diff
@@ -0,0 +1,227 @@
+from abc import ABCMeta
+from io import BytesIO
+from PIL import Image
+
+import numpy
+from six import PY3, add_metaclass
+
+from ._image import window_batch_bchw
+from . import ExpectsAxisLabels, SourcewiseTransformer
+from .. import config
+
+
+@add_metaclass(ABCMeta)
+class ImagesFromBytes(SourcewiseTransformer):
+    """Load from a stream of image-formatted data represented as bytes.
+
+    Parameters
+    ----------
+    data_stream : instance of :class:`AbstractDataStream`
+        The wrapped data stream.
+    which_sources : tuple of str
+        Which sources to apply the mapping to.
+    color_mode : str, optional
+        Mode to pass to PIL for color space conversion. Default is RGB.
+        If `None`, no coercion is performed.
+
+    Notes
+    -----
+    Images are returned as NumPy arrays converted from PIL objects.
+    If `color_mode` is RGB, then the array is transposed from the
+    `(height, width, channel)` dimension layout native to PIL to the
+    `(channel, height, width)` layout that is pervasive in the world of
+    convolutional networks.
+
+    This SourcewiseTransformer supports streams returning single examples
+    as `bytes` objects (`str` on Python 2.x) as well as streams that
+    return iterables containing such objects. In the case of an
+    iterable, a list of loaded images is returned.
+
+    """
+    def __init__(self, data_stream, which_sources, color_mode='RGB'):
```

Comment:
There are some kwargs that would be worth passing up the inheritance chain (namely `iteration_scheme`, which should always be `None`, and `axis_labels`).